### PR TITLE
Fix #4768: Parameter resolution via DashboardCardSeries in embedded multiline series

### LINKED ISSUE #4768: Multiple line series is not displayed in embedding
Version: 0.23.1

When I put multiple line series on a dashboard and embed it into an application,  the API, 

`GET /api/embed/dashboard/:token/dashcard/:dashcard-id/card/:card-id` 

returns 200 for the one series(card), but returns 400 for the rest.

### PR DESCRIPTION
**Pull request by alex-mcleod**
Proposed fix for #4768. Pretty new to Clojure and Metabase so there may be a more sensible way to do this. Cheers for any feedback.

### CODE REVIEW

File: src/metabase/api/embed.clj [hunk 1]
```diff
@@ -27,11 +27,14 @@
...
              [dashboard :refer [Dashboard]]
-             [dashboard-card :refer [DashboardCard]]]
+             [dashboard-card :refer [DashboardCard]]
+             [dashboard-card-series :refer [DashboardCardSeries]]]
             [metabase.util :as u]
             [metabase.util
              [embed :as eu]
              [schema :as su]]
+             [metabase
+              [db :as mdb]]
```
Comment (camsaul) [inline src/metabase/api/embed.clj]: this indentation is off

Comment (camsaul) [reply src/metabase/api/embed.clj]: also we try to keep all our requirements ordered, as you can see. So this should go further up in the file. It should just be `[metabase.db :as mdb`]

File: src/metabase/api/embed.clj [hunk 2]
```diff
@@ -164,11 +167,31 @@
...
+            )
+          )
+            :when         (= (:card_id param-mapping) card-id)
+            :let          [param (get param-id->param (:parameter_id param-mapping))]
+            :when         param
+        ]
+      (assoc param :target (:target param-mapping))
+    )
+  )
+)
```
Comment (camsaul) [inline src/metabase/api/embed.clj]: Hanging parens on their own lines is a Lisp no-no. Take a look at the rest of the file to get a sense of how the code should be styled

**Review by camsaul — CHANGES_REQUESTED** [review]
this fix might work. The code style could use some work however; as-is it makes reading/reviewing the code a bit difficult. Either way we'd want to add a test for this before merging

File: src/metabase/api/embed.clj (hunk 2 above)
Comment (camsaul) [reply src/metabase/api/embed.clj]: it doesn't look like you actually changed the last line so what's the reason behind changing all the parens like that?

**Comment by camsaul** [comment]
Thanks for your contribution @alex-mcleod. This fix looks like it might work so I'll look into it a bit more and see it that is indeed the case.

**Comment by camsaul** [comment]
Hey @alex-mcleod, I opened a new PR, #5985, which uses an approach similar to yours and adds a test as well. Thanks for the help investigating this issue. 🎉

**Comment by alex-mcleod** [comment]
@camsaul no worries, thanks for taking a look!

**Pull request closed without merging**